### ISSUE TITLE
Separate branch + pull request

### ISSUE THREAD

**Issue by poppyseedDev** [blockanalitica/ajna#31]
I propose that when committing changes to the codebase, the flow of contributing is to create a separate branch and then create a pull request, so that the commits to the main branch are less often and can be evaluated. 

---

For this purpose I purpose adding a `CONTRIBUTING.md` file something like this:

# Contributing Guidelines

Welcome to the [Project Name] project! We're excited that you're interested in contributing. Before you get started, please take a moment to read the following guidelines to ensure a smooth and collaborative experience.

## Table of Contents
- [Code of Conduct](#code-of-conduct)
- [How Can I Contribute?](#how-can-i-contribute)
- [Getting Started](#getting-started)
- [Development Workflow](#development-workflow)
- [Submitting a Pull Request](#submitting-a-pull-request)
- [Feedback and Discussions](#feedback-and-discussions)

## Code of Conduct

We expect all contributors to adhere to our [Code of Conduct](CODE_OF_CONDUCT.md). Please make sure to read it thoroughly and follow it in all your interactions with the community.

## How Can I Contribute?

There are several ways you can contribute to the project:

- Report bugs or suggest new features by opening [issues](https://github.com/yourusername/yourproject/issues).
- Implement new features or fix bugs by creating [pull requests](https://github.com/yourusername/yourproject/pulls).
- Improve the project's documentation.
- Participate in discussions and help resolve [issues](https://github.com/yourusername/yourproject/issues) and [pull requests](https://github.com/yourusername/yourproject/pulls).

We appreciate all contributions, big or small!

## Getting Started

To contribute to this project, follow these steps:

1. Fork the repository to your own GitHub account.
2. Clone the forked repository to your local machine.
3. Set up the project on your local machine by following the instructions in the `README.md` file.
4. Create a new branch for your contribution.

## Development Workflow

- Make sure to pull the latest changes from the `main` branch before starting your work.
- Write clean, well-documented, and tested code.
- Commit your changes in logical and granular commits.
- Push your changes to your forked repository.
- Regularly sync your forked repository with the upstream repository (the original repository you forked from) to stay up to date.

## Submitting a Pull Request

When you're ready to submit your changes, follow these steps:

1. Make sure your forked repository is up to date with the latest changes from the upstream repository.
2. Create a pull request (PR) from your branch to the `main` branch of the upstream repository.
3. Provide a descriptive title and a clear summary of your changes in the PR description.
4. Ensure that all automated tests pass successfully.
5. Be prepared to address feedback and make necessary changes if requested by the maintainers.
6. Once your PR is approved, it will be merged into the main repository.

## Feedback and Discussions

We welcome feedback, suggestions, and discussions related to the project. You can participate in the following ways:

- Join the [official community chat](https://example.com/chat) to interact with other contributors and maintainers.
- Open [issues](https://github.com/yourusername/yourproject/issues) for reporting bugs or suggesting improvements.
- Participate in discussions on existing [issues](https://github.com/yourusername/yourproject/issues) or [pull requests](https://github.com/yourusername/yourproject/pulls).

Thank you for your interest in contributing to the [Project Name] project! Your contributions are greatly appreciated.
